src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useState } from 'react';
+import { Person } from '../../types/Person';
+import React from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../../data/people';
+
+type Props = {
+  onSelected: (person: Person) => void;
+  delay: number;
+  inputIsFocused: boolean;
+  filteredList: Person[];
+  appliedQuery: string;
+  setAppliedQuery: (arg: string) => void;
+  setSelectedPerson: (person?: Person) => void;
+};
+
+export const Dropdown: React.FC<Props> = React.memo(
+  ({
+    onSelected,
+    delay,
+    filteredList,
+    setAppliedQuery,
+    appliedQuery,
+    setSelectedPerson,
+  }) => {
+    const [query, setQuery] = useState('');
+    const [showSuggestions, setShowSuggestions] = useState(false);
+
+    const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+    const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSelectedPerson();
+
+      setQuery(event.target.value);
+      applyQuery(event.target.value);
+
+      peopleFromServer.filter(person =>
+        person.name
+          .toLocaleLowerCase()
+          .includes(appliedQuery.toLocaleLowerCase()),
+      );
```

Comment:
not sure what are you trying to do here
filter is non-mutating method so you need to assign the result of its execution to variable. and I'm not sure that you need to filter people in this function